hedera-node/hedera-app/src/main/java/com/hedera/node/app/config/ConfigProviderImpl.java

```diff
@@ -34,11 +36,9 @@ public class ConfigProviderImpl implements ConfigProvider {
 
     private final PropertySource propertySource;
 
-    private Configuration configuration;
+    private final AtomicReference<VersionedConfiguration> configuration;
```

Comment:
I think an `AtomicReference` is not needed here. Making `configuration` `volatile` should be sufficient IMO, but I am definitely not an expert in these things.